File: backend/app/schemas/evaluation.py
from pydantic import BaseModel, Field


class CreateEvaluationRequest(BaseModel):
    score_name: str = Field(default="answer_quality", description="评分维度")
    score_value: float = Field(ge=0, le=10, description="评分值（0-10）")
    score_comment: str | None = None


class EvaluationInfo(BaseModel):
    id: int
    consultation_id: str
    trace_id: str
    score_name: str
    score_value: float
    data_type: str
    comment: str | None
    evaluated_by: str | None
    created_at: str


class EvaluationListData(BaseModel):
    items: list[EvaluationInfo]
    total: int
    page: int
    page_size: int

File: frontend/src/api/evaluation.ts
import client from "./client";
import type { ApiResponse } from "../types";

export interface EvaluationListData {
  items: {
    id: number;
    consultation_id: string;
    trace_id: string;
    score_name: string;
    score_value: number;
    data_type: string;
    comment: string | null;
    evaluated_by: string | null;
    question: string;
    created_at: string;
  }[];
  total: number;
  page: number;
  page_size: number;
}

export const evaluationApi = {
  list: (params: {
    page?: number;
    page_size?: number;
    score_name?: string;
    consultation_id?: string;
    evaluated_by?: string;
  }) =>
    client.get<ApiResponse<EvaluationListData>>("/evaluations", { params }),

  listScoreNames: () =>
    client.get<ApiResponse<string[]>>("/evaluations/score-names"),
};
